src/amo/components/LandingPage.js

```diff
@@ -0,0 +1,118 @@
+import classNames from 'classnames';
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { asyncConnect } from 'redux-connect';
+import { connect } from 'react-redux';
+
+import AddonsCard from 'amo/components/AddonsCard';
+import Link from 'amo/components/Link';
+import { loadLandingAddons } from 'amo/utils';
+import translate from 'core/i18n/translate';
+
+import './LandingPage.scss';
+
+
+export class LandingPageBase extends React.Component {
+  static propTypes = {
+    addonType: PropTypes.string.isRequired,
+    featuredAddons: PropTypes.array,
+    highlyRatedAddons: PropTypes.array,
+    popularAddons: PropTypes.array,
+    i18n: PropTypes.object.isRequired,
+  }
+
+  contentForType(addonType) {
+    const { i18n } = this.props;
+
+    const contentForTypes = {
+      extension: {
+        featuredHeader: i18n.gettext('Featured extensions'),
+        featuredFooter: (
+          <Link className="LandingPage-more-link" to="test">
+            {i18n.gettext('More featured extensions')}
+          </Link>
+        ),
+        popularHeader: i18n.gettext('Most popular extensions'),
+        popularFooter: (
+          <Link className="LandingPage-more-link" to="test">
+            {i18n.gettext('More popular extensions')}
+          </Link>
+        ),
+        highlyRatedHeader: i18n.gettext('Top rated extensions'),
+        highlyRatedFooter: (
+          <Link className="LandingPage-more-link" to="test">
+            {i18n.gettext('More highly rated extensions')}
+          </Link>
+        ),
+      },
+      theme: {
+        featuredHeader: i18n.gettext('Featured themes'),
+        featuredFooter: (
+          <Link className="LandingPage-more-link" to="test">
+            {i18n.gettext('More featured themes')}
+          </Link>
+        ),
+        popularHeader: i18n.gettext('Most popular themes'),
+        popularFooter: (
+          <Link className="LandingPage-more-link" to="test">
+            {i18n.gettext('More popular themes')}
+          </Link>
+        ),
+        highlyRatedHeader: i18n.gettext('Top rated themes'),
+        highlyRatedFooter: (
+          <Link className="LandingPage-more-link" to="test">
+            {i18n.gettext('More highly rated themes')}
+          </Link>
+        ),
+      },
+    };
+
+    if (contentForTypes[addonType]) {
+      return contentForTypes[addonType];
+    }
+
+    throw new Error(`No LandingPage content for addonType: ${addonType}`);
+  }
+
+  render() {
+    const {
+      addonType, featuredAddons, highlyRatedAddons, popularAddons,
+    } = this.props;
+
+    const html = this.contentForType(addonType);
+
+    return (
+      <div className={classNames('LandingPage', `LandingPage-${addonType}`)}>
+        <AddonsCard addons={featuredAddons} className="FeaturedAddons"
```

Comment:
Yeah I thought that too and was gonna do it in a tidy-up bug to avoid too many components and more tests in this PR, but I'll do it here if you're cool with that.